src/event.rs

```diff
@@ -386,6 +396,7 @@ impl<N: Notify> Processor<N> {
 
             // Scope needed to that hide_cursor isn't borrowed after the scope
             // ends.
+            let mut scrolled_lines = 0.0;
```

Comment:
Instead of starting from zero every time scroll events arrive, we should try and persist it. We might want this if we end up supporting smooth visual scrolling, and it may actually affect behavior regardless.

One option would be putting the `scrolled_lines` property onto the mouse state struct.